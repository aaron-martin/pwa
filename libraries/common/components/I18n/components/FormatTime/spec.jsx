/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import I18n from '../../index';

describe.skip('<FormatTime />', () => {
  let renderedElement;
  const testLocales = {
    greeting: 'Hello {time}',
  };
  const timestamp = new Date('Dec 25, 1999 04:25:45').getTime();
  const formattedTime = '4:25:45 AM';
  const format = 'medium';
  const langCode = 'en-US';

  /**
   * Renders the component.
   * @param {Object} props The component props.
   */
  const renderComponent = (props) => {
    renderedElement = mount((
      <I18n.Provider {...props}>
        <div>
          <span className="only-time">
            <I18n.Time timestamp={timestamp} format={format} />
          </span>
          <span className="text-with-time">
            <I18n.Text string="greeting">
              <I18n.Time forKey="time" timestamp={timestamp} format={format} />
            </I18n.Text>
          </span>
        </div>
      </I18n.Provider>
    ));
  };

  beforeEach(() => {
    renderComponent({
      lang: langCode,
      locales: testLocales,
    });
  });

  describe('Given the component was mounted to the DOM', () => {
    it('should match snapshot', () => {
      expect(renderedElement).toMatchSnapshot();
    });

    it('should render formatted time', () => {
      const text = renderedElement.find('.only-time').text();
      expect(text).toBe(formattedTime);
    });

    it('should render within translated text', () => {
      const text = renderedElement.find('.text-with-time').text();
      expect(text).toBe(`Hello ${formattedTime}`);
    });
  });
});
