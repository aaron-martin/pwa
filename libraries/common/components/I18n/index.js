/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import I18nProvider from './components/I18nProvider';
import Placeholder from './components/Placeholder';
import Translate from './components/Translate';
import FormatDate from './components/FormatDate';
import FormatTime from './components/FormatTime';
import FormatPrice from './components/FormatPrice';

export default {
  Provider: I18nProvider,
  Text: Translate,
  Date: FormatDate,
  Time: FormatTime,
  Placeholder,
  Price: FormatPrice,
};
