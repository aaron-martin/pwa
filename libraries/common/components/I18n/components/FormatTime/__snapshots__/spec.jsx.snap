// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatTime /> Given the component was mounted to the DOM should match snapshot 1`] = `
<I18nProvider
  lang="en-US"
  locales={
    Object {
      "greeting": "Hello {time}",
    }
  }
>
  <div>
    <span
      className="only-time"
    >
      <FormatTime
        format="medium"
        timestamp={946092345000}
      >
        <span>
          4:25:45 AM
        </span>
      </FormatTime>
    </span>
    <span
      className="text-with-time"
    >
      <Translate
        className=""
        params={Object {}}
        string="greeting"
      >
        <span
          className=""
        >
          Hello 
          <FormatTime
            forKey="time"
            format="medium"
            key=".0"
            timestamp={946092345000}
          >
            <span>
              4:25:45 AM
            </span>
          </FormatTime>
        </span>
      </Translate>
    </span>
  </div>
</I18nProvider>
`;
